collector/netdev_freebsd.go

```diff
@@ -45,39 +45,35 @@ func (c *netDevCollector) Update(ch chan<- prometheus.Metric) (err error) {
 	if err != nil {
 		return fmt.Errorf("couldn't get netstats: %s", err)
 	}
-	for direction, devStats := range netDev {
-		for dev, stats := range devStats {
-			for t, value := range stats {
-				key := direction + "_" + t
-				if _, ok := c.metrics[key]; !ok {
-					c.metrics[key] = prometheus.NewCounterVec(
-						prometheus.CounterOpts{
-							Namespace: Namespace,
-							Subsystem: netDevSubsystem,
-							Name:      key,
-							Help:      fmt.Sprintf("%s %s from getifaddrs().", t, direction),
-						},
-						[]string{"device"},
-					)
-				}
-				v, err := strconv.ParseFloat(value, 64)
-				if err != nil {
-					return fmt.Errorf("invalid value %s in netstats: %s", value, err)
-				}
-				c.metrics[key].WithLabelValues(dev).Set(v)
+	for dev, devStats := range netDev {
+		for key, value := range devStats {
+                        desc, ok := c.metricDescs[key]
+                        if !ok {
+                                desc = prometheus.NewDesc(
+                                        prometheus.BuildFQName(
+                                                Namespace, subsystem, key),
```

Comment:
There's no 80-char rule with code here, so IMO this would actually be more readable if args were single-line:

```go
desc = prometheus.NewDesc(
	prometheus.BuildFQName(Namespace, subsystem, key),
	fmt.Sprintf("%s from getifaddrs().", key),
	[]string{"device"},
	nil,
)
```